tools/codegen/gen_functionalization_type.py

```diff
@@ -318,13 +325,13 @@ def emit_inplace_functionalization_body(
     if str(f.func.name) == 'copy_':
         functional_call_str = """\
             auto tmp_intermediate = at::_ops::to_other::call(src_, self_, non_blocking, false, c10::nullopt);
-            tmp_output = at::_ops::expand_as::call(tmp_intermediate, self_);"""
+            tmp_output = at::_ops::expand_copy::call(tmp_intermediate, self_.sizes(), false);"""
```

Comment:
We don't have an `expand_as_copy()` ?

Reply:
Yeah this was a bit annoying. We don't have `expand_as_copy` because `expand_as` is a `CompositeImplicitAutograd` op, and at least right now, we don't have `_copy()` variants for `CompositeImpliciitAutograd` views.

We could add them... but they're kinda unnecessary for functionalization because we want decompositions to run before the functionalization pass anyway.

Well, they're unnecessary except for examples like the case above, where we want to do special handling for `copy_()` to call a view op that happens to be composite.

One thing I tried to do was originally was to leave the `expand_as` call, and arange for functionalization to run when it decomposes.  I had some trouble trying to get that to work, so this seemed simpler (manually writing the decomposition into `expand_copy`)

Reply:
That makes sense. This new version sounds better.